tests/console/snapper_cleanup_timeline.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that secure boot is set as expected.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Configuration", "Configure snapper.");
+    foreach my $param (keys %{$test_data->{snapper_config}}) {
+        assert_script_run("snapper set-config $param=$test_data->{snapper_config}->{$param}");
+    }
+
+    record_info("Check quota", "Verify that the percentage of root filesystem quota is less than 50%");
+    my $disk_space   = script_output("df -h /  | awk \'NR==2 {print \$4}\'");
+    my $qgroup_space = script_output("btrfs qgroup show -F --sync / | awk \'NR==3 {print \$2}\'");
+    die "Root filesystem quota take more than 50% of root disk space" if ($qgroup_space >= 0.5 * $disk_space);
+
+    record_info("Space > 20%", "Ensure that free space is more than 20%. Create a snapshot with timeline cleanup algorithm and make sure that it is not erased by the algorithm.");
+    $disk_space = script_output("df -h /  | awk \'NR==2 {print \$5}\'");
+    die "Used disk space is more than 80%" if ($disk_space >= 80);
+    assert_script_run("snapper create --description \"timeline\" --cleanup-algorithm timeline");
+    assert_script_run("snapper cleanup timeline");
+    assert_script_run("snapper ls | grep timeline");
+
+    record_info("Space < 20%", "Fill up disk space and make sure that the created snapshot gets erased by timeline cleanup algorithm.");
+    assert_script_run("dd if=/dev/urandom of=/tmp/blob bs=1M count=30500", timeout => 900);
+    $disk_space = script_output("df -h /  | awk \'NR==2 {print \$5}\'");
+    die "Used disk space is less than 80%" if ($disk_space <= 80);
+    assert_script_run("snapper cleanup timeline");
+    my $is_clean = script_run "snapper ls | grep timeline";
```

Comment:
Be careful with `script_run`, as it returns return code, so in case everything is good, `0` is returned, which corresponds to false value in perl.

Reply:
Indeed, if grep can't find "timeline", script_output will return error ("1") which means that there is no timeline snapshot, so is_clean=1. Here is a run where I skipped filling up the space and timeline snapshot was not erased and test fails: http://falafel.suse.cz/tests/746 . I have renamed the variable to is_snapshot_erased , in order to make it more comprehensible.

Reply:
You can also write something like `$is_clean = (script_run "snapper ls | grep timeline" != 0)` so that it's clear what we expect and to which value we convert it.

Reply:
TBH, this doesn't seem clearer to me. I have tried similar script and seems that, in case of a successful grep, script_run will return 0 but due to "!=0", $is_clean will be undefined.